src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -209,20 +136,42 @@ namespace
 int constexpr UNTOUCHED_NODE = -1;
 } // namespace
 
-template <typename MemorySpace>
-class GenerateHierarchyFunctor
+template <typename ExecutionSpace, typename Primitives, typename Nodes>
+inline void initializeSingleLeafNode(ExecutionSpace const &space,
+                                     Primitives const &primitives,
+                                     Nodes const &leaf_nodes)
+{
+  using Access = AccessTraits<Primitives, PrimitivesTag>;
+
+  Kokkos::parallel_for(
+      "ArborX::BVH::BVH::initialize_single_leaf",
+      Kokkos::RangePolicy<ExecutionSpace>(space, 0, 1), KOKKOS_LAMBDA(int i) {
+        Box bbox{};
+        expand(bbox, Access::get(primitives, 0));
+        leaf_nodes(0) = Details::makeLeafNode(0, std::move(bbox));
```

Comment:
Can you please elaborate on why `std::move` is needed here?

Reply:
It is not critical and without effect at the moment because `Box` is trivially-copyable.
It is defensive since I know the code may become templated on the bounding volume type.  It is only applied on a single leaf node so I can drop I you really don't like it.

Reply:
`makeLeafNode` already has `std::move` inside. Is this a duplication of already existing defensive approach?

Reply:
The function takes by value